testing/scenario_app/lib/main.dart

```diff
@@ -115,17 +119,33 @@ Future<Map<String, dynamic>> _getJson(Uri uri) async {
 }
 
 void _onBeginFrame(Duration duration) {
-  _currentScenario.onBeginFrame(duration);
+  _currentScenario?.onBeginFrame(duration);
 }
 
 void _onDrawFrame() {
-  _currentScenario.onDrawFrame();
+  _currentScenario?.onDrawFrame();
+
+  if (!_didReceiveScenarioFromPlatform) {
+    return;
+  }
+  _sceneCompletedFuture?.timeout(Duration.zero, onTimeout: () {});
```

Comment:
oh I see why TextPlatformViewActivity.java was so complicated. This is a signal for the second post-setting-scenario frame? Doing this across 2 sides is really complicated. Just get the renderer from the TextPlatformViewActivity and add a addIsDisplayingFlutterUiListener.